### DISCUSSION TITLE
Debugging in VS Code

### DISCUSSION THREAD

**Discussion by Phunction1** [krvajal/vscode-fortran-support#915] [General]
Hi, I am working on getting Fortran to work with VS Code, I have the Modern Fortran extension installed, c++, code runner, gdb.
I am able to run the Fortran code fine, but when I want to debug it asks be to choose a debugger.
I setup the launch.json as per the example provided.
VS Code does work fine running and debugging C/C++ code so not sure what I am missing for Fortran?

**Comment by gnikit** [comment]
Can you provide some more details:
- Modern Fortran verion:
- Operating system
- any options you've passed to Modern Fortran
- Reproducible steps you followed to get vscode to compile and debug your program

**Reply by Phunction1** [reply]
MF version 3.2.0
Windows 10
I have not changed any options for MF
Basically started a Fortran file, when to Run and Debug, and it popups the little drop down with node.js and chrome app choices, and install an extension for Fortran
I should mention that I am trying to work with Fortran 77

**Reply by gnikit** [reply]
for Windows you will have to use GFortran and GDB if you wish to run a debugger (using WSL seems to work well for a lot of users).
Unfortunately, Microsoft/Intel, does not offer Debug adaptors for the Intel compilers (AFAIK).

A debug adaptor is the line in `launch.json`: `"type": "cppdbg",`. Have a look here for more: https://code.visualstudio.com/docs/cpp/config-mingw

**Reply by albertziegenhagel** [reply]
> Unfortunately, Microsoft/Intel, does not offer Debug adaptors for the Intel compilers (AFAIK).

It is actually possible to use the debug adaptors that Intel ships for Visual Studio in Visual Studio Code as well, but it requires a bit of manual setup. I have written down the procedure to set it up here: https://gist.github.com/albertziegenhagel/6431811950864bd0009b6a1fa78e7f2b.

I would be more than happy if anyone tries that and let's me know if anything is unclear or whether there are any other issues.

**Reply by gnikit** [reply]
@albertziegenhagel what you posted seems promising, let me split this into a separate `Show and Tell` Discussion and carry the relevant conversation there.

**Comment by Phunction1** [comment]
Yes, I have that in the json file, I also tried adding this at the top:
"miDebuggerPath": "c:\\mingw\\bin\\gdb.exe",

**Comment by Phunction1** [comment]
I have proceeded a little farther, now I am stuck on this line in json:
"program": "${workspaceFolder}\\test.exe",
It complains that is is not an executable or that it does not exist. I double checked and it is there and I can run it from a terminal.

I originally tried the a.out that the example had but no go.

**Reply by gnikit** [reply]
I don't have enough info to assist with debugging but if you install the pre-release version of the extension and set the log level to debug, then a bunch of log messages will be printed in the OUTPUT pane of VS Code under the _Modern Fortran_ dropdown. Pasting those here would clarify a lot of things.

Alternatively, you can upload a zipped Minimal Working Example of your workspace (files, vscode scripts, etc.) and I can have a look.

**Reply by gnikit** [reply]
In the vscode settings you should have an options with the following JSON ID (you can also copy-paste it into your settings):

```json
  "fortran.logging.level": "Debug",
```

**Reply by Phunction1** [reply]
When I switched to the per-release, it not longer recognized my .f files, everything came up as syntax errors so I switched back to the release version.

**Reply by gnikit** [reply]
That is not consistent with the continuously tested behaviour of the extension. .f files are registered as Fortran at the core of the extension, that hasn't changed since 2017.

I am afraid without a complete and minimal set of files, settings, installed vscode extensions, etc. that are able to reproduce your problem in a workspace, I won't be able to provide any more help, since I would be randomly guessing.

You might want to have a look at similar Discussions and issues in the repository as to the type of information you need to provide.
